### PR TITLE
Mobile support  (server side)

### PR DESCRIPTION
**Pull request by margusp2heathmont**
Screenshot 2023-05-10 at 00 15 27

# This enables a adaptive API for GUI component programming.

CRUD.

# Helper components for CSS media queries and client side width detection.

Screenshot 2023-05-09 at 23 50 16

#  This introduces Moon.UIPattern.API.AddResource

That is automatically adaptable per device.

Maybe this should be github.com/moon-elixir/ui-pattern/?

## On mobile (BottomSheet)
Screenshot 2023-05-10 at 13 19 08

## On  desktop (Modal)
Screenshot 2023-05-10 at 13 18 33

Some reasoning:
1. In user-interface we might want to make decisions only with CSS media queries? Or on the level of what data is fetched per device type. In addition, providing components for generic scenarios helps to unify codebase and quality.

### CODE REVIEW

File: assets/js/set-is-mobile.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import getCookie from "./cookies/get-cookie";
+import setCookie from "./cookies/set-cookie";
+
+export default function setIsMobile() {
+    let screenWidth = window.innerWidth
+        || document.documentElement.clientWidth
+        || document.body.clientWidth
+        || 0;
+
+    let isMobile = screenWidth < 1024;
```
Comment (margusp2heathmont) [inline assets/js/set-is-mobile.js:10]: Same could be done with UserAgent detection, but this is better imho.

Comment (phgrey) [reply assets/js/set-is-mobile.js:10]: Sorry, but I'm really sure we have to move all this functionality to AffiliateFlask project.
For some reasons - not sure this is the best way to support different view patterns.
Will be glad to see this functionality in a "howto" page format.

**Comment by phgrey** [comment]
Sorry, but I'm really sure we have to move all this functionality to AffiliateFlask project.
For some reasons - not sure this is the best way to support different view patterns.
Will be glad to see this functionality in a "howto" page format.

@dkireev please confirm that right now we're not going to add UIPattern / Responsive / Layout components to MoonDS

**Comment by margusp2heathmont** [comment]
Hey!

1. Do you agree, that fundamentally you have (1) client browser device detection (css or js) (2) server side browser detection?
2. Do you agree, that my PR-s provide technical solutions for either available way?
3. Do you agree, that both of those ways should be described by Moon, and Moon should have those components?

Now, next group of important questions is:

4. Do you agree, that server side browser detection enables limitation of data fetch?

**Comment by margusp2heathmont** [comment]
Closing because better solution found (responsive CSS only)

**Pull request closed without merging**